VMEncryption/main/EncryptionSettingsUtil.py

```diff
@@ -259,11 +262,42 @@ def post_to_wireserver(self, data):
         else:
             raise Exception("no response from encryption settings update request")
 
-    def clear_encryption_settings(self):
-        """ Clear settings by calling DisableEncryption operation via wire server"""
-        data = {"DiskEncryptionDataVersion": "2.0",
+    def clear_encryption_settings(self, disk_util):
+        """ 
+        Clear settings by calling DisableEncryption operation via wire server
+
+        finds all azure data disks and clears their encryption settings
+        """
+
+        self.logger.log("Clearing encryption settings for all data drives")
+
+        data_disk_controller_ids_and_luns = disk_util.get_all_azure_data_disk_controller_and_lun_numbers()
+
+        # validate machine name string or use empty string
+        machine_name = socket.gethostname()
+        if re.match('^[\w-]+$', machine_name) is None:
+            machine_name = ''
+
+        def controller_id_and_lun_to_settings_data(scsi_controller, lun_number):
+            return {
+                "ControllerType": "SCSI",
+                "ControllerId": scsi_controller,
+                "SlotId": lun_number,
+                "Volumes": [{
+                    "VolumeType": "DataVolume",
+                    "ProtectorFileName": "nullProtector.bek",
+                    "SecretTags": self._dict_to_name_value_array({
+                        "DiskEncryptionKeyFileName": "nullProtector.bek",
+                        "MachineName": machine_name})
+                    }]
+                }
+
+        data_disks_settings_data = [ controller_id_and_lun_to_settings_data(scsi_controller, lun_number)
```

Comment:
what about OS disk on IDE controller?

Reply:
We don't allow disable when the OS is encrypted so I left it out. But, on second thought I will add that in - in case somebody somehow encrypts a partition of the OS disk with volume type data.